pkg/acsengine/defaults.go

```diff
@@ -82,6 +82,14 @@ func setOrchestratorDefaults(cs *api.ContainerService) {
 		if a.OrchestratorProfile.KubernetesConfig.NetworkPolicy == "" {
 			a.OrchestratorProfile.KubernetesConfig.NetworkPolicy = DefaultNetworkPolicy
 		}
+		if a.OrchestratorProfile.KubernetesConfig.ClusterSubnet == "" {
+			if a.OrchestratorProfile.IsVNETIntegrated() {
+				// When VNET integration is enabled, all masters, agents and pods share the same large subnet.
```

Comment:
(2) wouldn't work for the case where user-specified ClusterSubnet != "". We would override it with the default. This has to be nested because the inner if check must run iff KubernetesConfig.ClusterSubnet == "".

I thought this logic is simple enough and refactoring it to another function would just make the code less readable.